packages/flare/python/cron_job_ingest_events.py

```diff
@@ -29,44 +33,133 @@ def main():
+    except Exception:
+        pass  # Adding logger in next PR
+
+    set_job_running(app=app, is_running=False)
 
 
-def get_from_value(app: client.Application) -> Optional[str]:
-    from_: Optional[str] = None
+def get_collection_value(app: client.Application, key: str) -> Optional[Any]:
     if KV_COLLECTION_NAME in app.service.kvstore:
         data = app.service.kvstore[KV_COLLECTION_NAME].data.query()
-        if len(data) > 1:
-            from_ = data[0]["value"]
-        else:
-            app.service.kvstore.delete(KV_COLLECTION_NAME)
+        for entry in data:
+            if entry["_key"] == key:
+                return entry["value"]
+
+    return None
+
+
+def get_next(app: client.Application, tenant_id: int) -> Optional[str]:
+    return get_collection_value(app=app, key="next_{}".format(tenant_id))
 
-    return from_
 
+def get_start_date(app: client.Application) -> Optional[str]:
+    return get_collection_value(app=app, key="start_date")
 
-def set_from_value(app: client.Application, next: Optional[str]):
+
+def get_current_tenant_id(app: client.Application) -> Optional[int]:
+    current_tenant_id = get_collection_value(app=app, key="current_tenant_id")
+    return int(current_tenant_id) if current_tenant_id else None
+
+
+def get_job_running(app: client.Application) -> Optional[bool]:
+    is_job_running = get_collection_value(app=app, key="is_job_running")
+    return bool(is_job_running) if is_job_running else None
+
+
+def create_collection(app: client.Application) -> None:
     if KV_COLLECTION_NAME not in app.service.kvstore:
         # Create the collection
         app.service.kvstore.create(
             name=KV_COLLECTION_NAME, fields={"_key": "string", "value": "string"}
         )
-        # Insert
+
+
+def save_start_date(app: client.Application, tenant_id: int) -> None:
+    current_tenant_id = get_current_tenant_id(app=app)
+    # If this is the first request ever, insert today's date so that future requests will be based on that
+    if not get_start_date(app):
+        app.service.kvstore[KV_COLLECTION_NAME].data.insert(
+            json.dumps(
+                {
+                    "_key": "start_date",
+                    "value": date.today().isoformat(),
+                }
+            )
+        )
+
+    # If the current tenant has changed, update the start date so that future requests will be based off today
+    # If you switch tenants, this will avoid the old tenant from ingesting all the events before today and the day
+    # that tenant was switched in the first place.
+    if current_tenant_id != tenant_id:
+        app.service.kvstore[KV_COLLECTION_NAME].data.update(
+            id="start_date", data=json.dumps({"value": date.today().isoformat()})
+        )
+
+
+def save_next(app: client.Application, tenant_id: int, next: Optional[str]) -> None:
+    # If we have a new next value, update the collection for that tenant to continue searching from that point
+    if next:
```

Comment:
Instead of indenting the whole function, we usually reverse the check and return early.

```python
if not next:
    return
rest of code...
```